### PR TITLE
remove locations, persons and use v2 to update status

### PR DESCRIPTION
**Pull request by wow-such-code**

### CODE REVIEW

File: src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy [hunk 1]
```diff
@@ -33,54 +30,36 @@ class QbicSampleTrackingService implements SampleTrackingService {
...
-        return Optional.of(response.body()).flatMap(DtoMapper::parseLocationOfJson)
-    }
-
-    private HttpResponse<String> requestSampleUpdate(URI requestURI, String locationJson) {
+    private HttpResponse<String> requestSampleUpdate(URI requestURI, String statusJson) {
         HttpRequest request = HttpRequest
                 .newBuilder(requestURI)
                 .header("Content-Type", "application/json")
-                .PUT(HttpRequest.BodyPublishers.ofString(locationJson))
+                .PUT(HttpRequest.BodyPublishers.ofString(statusJson))
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy:53]: do we need to request a sample update? I guess before this was to verify if a valid location makes the update?

File: src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy [hunk 2]
```diff
@@ -93,112 +72,28 @@ class QbicSampleTrackingService implements SampleTrackingService {
...
     @Override
-    void updateSampleLocation(String sampleCode, Location location, String status, Instant timestamp) throws SampleUpdateException {
-        String locationJson = DtoMapper.createJsonFromLocationWithStatus(location, status, timestamp)
-        HttpResponse<String> response = requestSampleUpdate(createSampleUpdateURI(sampleCode), locationJson)
+    void updateSampleStatus(String sampleCode, String status, Instant timestamp) throws SampleUpdateException {
+        String statusJson = DtoMapper.createJsonFromStatus(status, timestamp)
+        HttpResponse<String> response = requestSampleUpdate(createSampleUpdateURI(sampleCode), statusJson)
         if (response.statusCode() != 200) {
             throw new SampleUpdateException("Could not update $sampleCode to ${location.getLabel()} - ${response.statusCode()} : ${response.headers()}: ${response.body()}")
         }
```
Comment (jenniferboedker) [inline src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy]: here you would also need to remove the location

File: src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy [hunk 3]
```diff
@@ -93,112 +72,28 @@ class QbicSampleTrackingService implements SampleTrackingService {
...
+            return JsonOutput.toJson(parameters)
         }
 
-        private static String mapToLocationDateTimeString(Instant timestamp) {
+        private static String mapToDateTimeString(Instant timestamp) {
             // the pattern mentioned here is dictated by the data-model-lib location object
             var dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm'Z'")
                     .withZone(ZoneId.from(UTC))
             return dateTimeFormatter.format(timestamp)
         }
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy]: You do not need this anymore as the new sample tracking API handles instants.

File: src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy [hunk 4]
```diff
@@ -93,112 +72,28 @@ class QbicSampleTrackingService implements SampleTrackingService {
...
-
-        protected static String createJsonFromLocationWithStatus(Location location, String status, Instant arrivalTime) {
-            Map locationMap = convertLocationToMap(location)
-            locationMap.put("arrival_date", mapToLocationDateTimeString(arrivalTime))
-            locationMap.put("sample_status", status)
-            return JsonOutput.toJson(locationMap)
+        protected static String createJsonFromStatus(String status, Instant arrivalTime) {
+            Map parameters = new HashMap()
+            parameters.add("status" : status)
+            parameters.add("validFrom" : mapToDateTimeString(arrivalTime))
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/samplestatus/reporter/services/QbicSampleTrackingService.groovy]:
```suggestion
            parameters.add("validFrom" : arrivalTime.toString())
```
As the new API circumvents the weird location time format, you can use the ISO format directly (provided by instant and automatically as UTC)

**Review by jenniferboedker — APPROVED** [review]

**Pull request merged**